src/services/events/event-manager.service.ts

```diff
@@ -1,35 +1,33 @@
 import 'reflect-metadata';
-
 import EventEmitter from 'events';
-import { Subject } from 'rxjs';
-
-import { OnEventInterface } from '../../decorators';
+import { Injectable, OnEventInterface } from '../../decorators';
 import { NativeEventsType } from '../../models';
-import { NATIVE_SERVICES } from '../../models/constants/native-services';
-import { Injector } from '../../models/dependency-injection/injector.service';
+import { DECORATORS } from '../../models/constants/decorators';
+import { OnProviderInit } from '../../models/interfaces/life-cycle.interface';
+import { Subjects } from '../subjects/subjects';
+import { SubjectEvent } from '../../models/interfaces/subject.interface';
 
 export interface EventPayload<TData> {
   event: string | NativeEventsType;
   timestamp: number;
   data: TData;
 }
 
-export interface LifeCycleEvent<TData> {
-  event: NativeEventsType;
-  data: TData;
-}
-
-export const LifeCycleEventEmitter = new Subject<LifeCycleEvent<unknown>>();
-
-export class EventManagerService {
+@Injectable()
+export class EventManagerService implements OnProviderInit {
   private emitter = new EventEmitter();
 
-  constructor() {
-    LifeCycleEventEmitter.subscribe((event: LifeCycleEvent<unknown>) => {
-      if (event) {
-        this.push(event.event, event.data);
-      }
-    });
+  onProviderInit() {
+    const listeners: OnEventInterface[] = Reflect.getMetadata(DECORATORS.metadata.events.ON_EVENT, EventManagerService) || [];
+    listeners.forEach(listener => this.register(listener));
+
+    for (const subject in Subjects) {
+      Subjects[subject].subscribe((event: SubjectEvent<unknown>) => {
+        if (event) {
+          this.push(event.event, event.data);
+        }
+      })
+    }
```

Comment:
`for...in` syntax has performance issues, this could be written as:
```suggestion
    for (const key in Object.keys(Subjects)) {
      Subjects[key].subscribe((event: SubjectEvent<unknown>) => {
        if (event) {
          this.push(event.event, event.data);
        }
      })
    }
```

Besides that, providing a way to `unsubscribe` would be appropriate to avoid memory leaks. Maybe we could implement onProviderDestroy and do both unsubscribes and removal of event listeners.